pkg/daemon/ceph/client/filesystem_test.go

```diff
@@ -166,3 +167,195 @@ func TestFilesystemRemove(t *testing.T) {
 	assert.True(t, dataDeleted)
 	assert.True(t, crushDeleted)
 }
+
+func TestFailAllStandbyReplayMDS(t *testing.T) {
+	executor := &exectest.MockExecutor{}
+	context := &clusterd.Context{Executor: executor}
+	fs := CephFilesystemDetails{
+		ID: 1,
+		MDSMap: MDSMap{
+			FilesystemName: "myfs1",
+			MetadataPool:   2,
+			Up: map[string]int{
+				"mds_0": 123,
+			},
+			DataPools: []int{3},
+			Info: map[string]MDSInfo{
+				"gid_123": {
+					GID:   123,
+					State: "up:active",
+					Name:  fmt.Sprintf("%s-%s", "myfs1", "a"),
+				},
+				"gid_124": {
+					GID:   124,
+					State: "up:standby-replay",
+					Name:  fmt.Sprintf("%s-%s", "myfs1", "b"),
+				},
+			},
+		},
+	}
+	executor.MockExecuteCommandWithOutputFile = func(command, outputFile string, args ...string) (string, error) {
+		logger.Infof("Command: %s %v", command, args)
+		if args[0] == "fs" {
+			if args[1] == "get" {
+				output, err := json.Marshal(fs)
+				assert.Nil(t, err)
+				return string(output), nil
+			}
+			if args[1] == "rm" {
+				return "", nil
+			}
+		}
+		if args[0] == "mds" {
+			if args[1] == "fail" {
+				return "", nil
+			}
+		}
+		return "", errors.Errorf("unexpected ceph command %q", args)
+	}
+
+	err := FailAllStandbyReplayMDS(context, AdminClusterInfo("mycluster"), fs.MDSMap.FilesystemName)
+	assert.Nil(t, err)
```

Comment:
When testing errors, it's more clear to use assert's `NoError` or `Error` functions.

```suggestion
	assert.NoError(t, err)
```

Reply:
This is good, but how can we also test that it called `fail` on the correct MDSes?

I might suggest taking a look here. `argsSeen` is an example of a pattern I have used to have the executor update some info about what commands it has issued so that I can verify in unit tests that the right commands are being called.
https://github.com/rook/rook/blob/795124b7a8d625df7726679941e4f9a18adf47b0/pkg/daemon/ceph/client/osd_test.go#L117

In this case, I would probably have something similar to below...
```go
var mdsFailCalls []string

// in the executor
// ...
if args[2] == "fail" {
    mdsFailCalls = append(mdsFailCalls, args[3])
}
// ...

// in a test
mdsFailCalls = []string{} // reset to empty before the test
err := FailAllStandbyReplayMDS(context, AdminClusterInfo("mycluster"), fs.MDSMap.FilesystemName)
assert.Nil(t, err)
assert.ElementsMatch(t, mdsFailCalls, []string{"myfs1-mds-b"}) // or whatever is appropriate... I think it's actually a GID and not the name